apps/server/src/modules/h5p-editor/controller/dto/ajax/post.body.params.ts

```diff
@@ -0,0 +1,72 @@
+import { BadRequestException, Injectable, PipeTransform, ValidationPipe } from '@nestjs/common';
+import { AnyFilesInterceptor } from '@nestjs/platform-express';
+import { ApiProperty } from '@nestjs/swagger';
+import { plainToClass } from 'class-transformer';
+import { IsArray, IsMongoId, IsOptional, IsString, validate } from 'class-validator';
+
+class LibrariesBodyParams {
+	@ApiProperty()
+	@IsArray()
+	@IsString({ each: true })
+	libraries!: string[];
+}
+
+class ContentBodyParams {
+	@ApiProperty()
+	@IsMongoId()
+	contentId!: string;
+
+	@ApiProperty()
+	@IsString()
+	@IsOptional()
+	field!: string; // Todo: Reason
+}
+
+class LibraryParametersBodyParams {
+	@ApiProperty()
+	@IsString()
+	libraryParameters!: string;
+}
+
+export type AjaxPostBodyParams = LibrariesBodyParams | ContentBodyParams | LibraryParametersBodyParams | undefined;
+
+@Injectable()
+export class AjaxPostBodyParamsTransformPipe implements PipeTransform {
+	async transform(value: AjaxPostBodyParams) {
+		if (value) {
+			let transformed: Exclude<AjaxPostBodyParams, undefined>;
+
+			if ('libraries' in value) {
+				transformed = plainToClass(LibrariesBodyParams, value);
+			} else if ('contentId' in value) {
+				transformed = plainToClass(ContentBodyParams, value);
+			} else if ('libraryParameters' in value) {
+				transformed = plainToClass(LibraryParametersBodyParams, value);
+			} else {
+				return undefined; // Todo Check this
+			}
+
+			const validationResult = await validate(transformed);
+			if (validationResult.length > 0) {
+				const validationPipe = new ValidationPipe();
+				const exceptionFactory = validationPipe.createExceptionFactory();
+				throw exceptionFactory(validationResult);
+			}
+
+			return transformed;
+		}
+
+		return undefined;
+	}
+}
+
+export const AjaxPostBodyParamsFilesInterceptor = AnyFilesInterceptor({
```

Comment:
H5P may or may not send files with the ajax request.
Since these files may be sent in two different fields, we cannot use Nest's default file interceptor (Only supports one field)
This interceptor is a specialization of the AnyFilesInterceptor that only accepts files with the two known fieldnames

Reply:
Found out about Nest's 'FileFieldsInterceptor', using that now